subsys/usb/class/audio/usb_audio_internal.h

```diff
@@ -0,0 +1,527 @@
+)
+
+/* Macro for getting required size to store values for supported features. */
+#define FEATURES_SIZE(dev, i) ((CH_CNT(dev, i) + 1) * (0x0000	\
++ IF_USB_AUDIO_PROP(dev, i, feature_mute,			\
+			    FEATURE_MUTE_SIZE)			\
++ IF_USB_AUDIO_PROP(dev, i, feature_volume,			\
+			    FEATURE_VOLUME_SIZE)		\
++ IF_USB_AUDIO_PROP(dev, i, feature_tone_control,		\
+			    FEATURE_TONE_CONTROL_SIZE)		\
++ IF_USB_AUDIO_PROP(dev, i, feature_graphic_equalizer,		\
+			    FEATURE_GRAPHIC_EQUALIZER_SIZE)	\
++ IF_USB_AUDIO_PROP(dev, i, feature_automatic_gain_control,	\
+			    FEATURE_AUTOMATIC_GAIN_CONTROL_SIZE)\
++ IF_USB_AUDIO_PROP(dev, i, feature_delay,			\
+			    FEATURE_DELAY_SIZE)			\
++ IF_USB_AUDIO_PROP(dev, i, feature_bass_boost,			\
+			    FEATURE_BASS_BOOST_SIZE)		\
++ IF_USB_AUDIO_PROP(dev, i, feature_loudness,			\
+			    FEATURE_LOUDNESS_SIZE)		\
+))
+
+#define GET_RES_HP(i) DT_PROP(DT_INST(i, COMPAT_HP), resolution)
+#define GET_RES_MIC(i) DT_PROP(DT_INST(i, COMPAT_MIC), resolution)
+#define GET_RES_HS_HP(i) DT_PROP(DT_INST(i, COMPAT_HS), hp_resolution)
+#define GET_RES_HS_MIC(i) DT_PROP(DT_INST(i, COMPAT_HS), mic_resolution)
+#define GET_RES(dev, i) GET_RES_##dev(i)
+
+#define SYNC_TYPE_HP(i)     3
+#define SYNC_TYPE_MIC(i)    DT_ENUM_IDX(DT_INST(i, COMPAT_MIC), sync_type)
+#define SYNC_TYPE_HS_HP(i)  3
+#define SYNC_TYPE_HS_MIC(i) DT_ENUM_IDX(DT_INST(i, COMPAT_HS), mic_sync_type)
+#define SYNC_TYPE(dev, i) (SYNC_TYPE_##dev(i) << 2)
+
+#define EP_SIZE(dev, i) \
+	((GET_RES(dev, i)/8) * CH_CNT(dev, i) * 48)
+
+/* *_ID() macros are used to give proper Id to each entity describing
+ * the device. Entities Id must start from 1 that's why 1 is added.
+ * Multiplication by 3 for HP/MIC comes from the fact that 3 entities are
+ * required to describe the device, 6 in case of HS.
+ */
+#define HP_ID(i) ((3*i) + 1)
+#define MIC_ID(i) ((3*(HEADPHONES_DEVICE_COUNT + i)) + 1)
+#define HS_ID(i) ((3*(HEADPHONES_DEVICE_COUNT + \
+		      MICROPHONE_DEVICE_COUNT)) + 6*i + 1)
+
+/* *_LINK() macros are used to properly connect relevant audio streaming
+ * class specific interface with valid entity. In case of Headphones this
+ * will always be 1st entity describing the device (Input terminal). This
+ * is where addition of 1 comes from. In case of Headphones thill will always
+ * be 3rd entity (Output terminal) - addition of 3.
+ */
+#define HP_LINK(i) ((3*i) + 1)
+#define MIC_LINK(i) ((3*(HEADPHONES_DEVICE_COUNT + i)) + 3)
+
+/**
+ * Addressable logical object inside an audio function.
+ * Entity is one of: Terminal or Unit.
+ * Refer to 1.4 Terms and Abbreviations from audio10.pdf
+ */
+struct usb_audio_entity {
+	enum usb_audio_cs_ac_int_desc_subtypes subtype;
+	u8_t id;
+};
+
+/**
+ * @warning Size of baInterface is 2 just to make it useable
+ * for all kind of devices: headphones, microphone and headset.
+ * Actual size of the struct should be checked by reading
+ * .bLength.
+ */
+struct cs_ac_interface_descriptor_header {
```

Comment:
please use shorter names, this one is 33 characters long, you can use _iface_ for interface and _desc_ for descriptor as abbreviation

Reply:
I will replace it with `cs_ac_if_descriptor` to be more or less consistent with what we have in the tree, see [usb_if_descriptor](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/usb/usb_common.h#L188).